fluss-common/src/test/java/com/alibaba/fluss/utils/concurrent/FutureUtilsTest.java

```diff
@@ -389,4 +397,143 @@ void testOrTimeout() {
+        CompletableFuture<String> failureFuture = new CompletableFuture<String>();
+        FutureUtils.ConjunctFuture<Collection<String>> failureConjunctFuture =
+                FutureUtils.combineAll(Arrays.asList(successFuture, failureFuture));
+
+        successFuture.complete("success");
+        failureFuture.completeExceptionally(new RuntimeException("mock runtime exception"));
+
+        assertThatFuture(failureConjunctFuture).isDone();
+        assertThatFuture(failureConjunctFuture)
+                .eventuallyFailsWith(ExecutionException.class)
+                .withCauseInstanceOf(RuntimeException.class)
+                .withMessageContaining("mock runtime exception");
+    }
+
+    @Test
+    void testWaitForAll() {
+        FutureUtils.ConjunctFuture<Void> emptyConjunctFuture =
+                FutureUtils.waitForAll(Collections.emptyList());
+        assertThatFuture(emptyConjunctFuture).isDone();
+        assertThatFuture(emptyConjunctFuture).eventuallySucceeds().isNull();
+
+        CompletableFuture<String> future1 = new CompletableFuture<String>();
+        CompletableFuture<String> future2 = new CompletableFuture<String>();
+        FutureUtils.ConjunctFuture<Void> conjunctFuture =
+                FutureUtils.waitForAll(
+                        Arrays.asList(future1, future2),
+                        (val, throwable) -> {
+                            if (throwable == null) {
+                                System.out.println(val + " Fluss");
+                            } else {
+                                throw new RuntimeException(throwable);
+                            }
+                        });
+
+        assertThatFuture(conjunctFuture).isNotDone();
+        assertThat(conjunctFuture.getNumFuturesTotal()).isEqualTo(2);
+        assertThat(conjunctFuture.getNumFuturesCompleted()).isZero();
+
+        future1.complete("Hello");
+        future2.complete("World");
+        assertThatFuture(conjunctFuture).eventuallySucceeds();
+
+        assertThatFuture(conjunctFuture).isDone();
+        assertThat(conjunctFuture.getNumFuturesTotal()).isEqualTo(2);
+        assertThat(conjunctFuture.getNumFuturesCompleted()).isEqualTo(2);
+
+        CompletableFuture<String> successFuture = new CompletableFuture<String>();
+        CompletableFuture<String> failureFuture = new CompletableFuture<String>();
+        FutureUtils.ConjunctFuture<Void> failureConjunctFuture =
+                FutureUtils.waitForAll(
+                        Arrays.asList(successFuture, failureFuture),
+                        (val, throwable) -> {
+                            if (throwable == null) {
+                                System.out.println(val + " Fluss");
+                            } else {
+                                throw new RuntimeException(throwable);
+                            }
+                        });
+
+        successFuture.complete("success");
```

Comment:
Do not complete the `successFuture` to verify the `waitForAll()` future fails once ONE of the futures fails.